codalab/rest/bundles.py

```diff
@@ -386,6 +389,56 @@ def _fetch_bundle_contents_info(uuid, path=''):
         'data': info
     }
 
+@put('/bundles/<uuid:re:%s>/netcat/<port:int>/' % spec_util.UUID_STR, name='netcat_bundle')
+def _netcat_bundle(uuid, port):
+    check_bundles_have_read_permission(local.model, request.user, [uuid])
+    bundle = local.model.get_bundle(uuid)
+    if bundle.state in State.FINAL_STATES:
+        abort(httplib.FORBIDDEN, 'Cannot netcat bundle, bundle already finalized.')
+    info = local.download_manager.netcat(uuid, port, request.json['message'])
+    return {
+        'data': info
+    }
+
+@post('/bundles/<uuid:re:%s>/netcurl/<port:int>/<path:re:.*>' % spec_util.UUID_STR, name='netcurl_bundle')
```

Comment:
Why doesn't curl subsume cat?

Reply:
the `curl` is specifically made to consume HTTP requests, and `cat` for arbitrary bytestrings. `cat` has to be separate because we add our own around the raw bytestring payload in order to pass it through bottle.